packages/core/src/walk.ts

```diff
@@ -212,16 +212,15 @@ export function walkDocument<T>(opts: {
             if (!activatedOn.skipped) {
               visitedBySome = true;
               enteredContexts.add(context);
-              const ignoreNextVisitorsOnNode = visitWithContext(
+              const { ignoreNextVisitorsOnNode, isDeleted } = visitWithContext(
                 visit,
                 resolvedNode,
                 context,
                 ruleId,
                 severity,
               );
-              if (ignoreNextVisitorsOnNode) {
-                break;
-              }
+              if (ignoreNextVisitorsOnNode) break;
+              if (isDeleted) return;
```

Comment:
need unit test for this case

Reply:
@Marshevskyy `delete` logic removed, use another better approach.